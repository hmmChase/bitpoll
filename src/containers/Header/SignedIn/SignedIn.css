.signed-in-state {
  display: flex;
  align-items: center;
  margin: 20px;
}

.log-in-out {
  position: relative;
}

.welcome-text {
  color: #000;
}

.verified-icon {
  margin-right: 2px;
}

.verifed-text {
  font-size: 0.7em;
  position: absolute;
  top: 25px;
  color: green;
}

.sign-out-btn {
  background: #3a3133;
  color: #fff;
  border: none;
  border-radius: 10px;
  height: 50px;
  display: flex;
  align-items: center;
  padding: 10px;
  margin-left: 20px;
}

.sign-out-btn:hover {
  background: #000;
}

