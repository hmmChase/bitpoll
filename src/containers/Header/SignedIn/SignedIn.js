import React from 'react';
import { connect } from 'react-redux';
import fontawesome from '@fortawesome/fontawesome';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faCheck from '@fortawesome/fontawesome-free-regular';
import './SignedIn.css';

const SignedIn = props => {
  return (
    <div className="signed-in-state">
      <div className="log-in-out">
        <p className="welcome-text">Welcome {props.displayName}</p>
        {props.isContributor && (
          <p className="verifed-text">
            <FontAwesomeIcon
              className="verified-icon"
              icon={['far', 'check-circle']}
            />verified contributor
          </p>
        )}
      </div>
      <div className="sign-out-btn" onClick={props.logOut}>
        <span>Log out</span>
      </div>
    </div>
  );
};

export const mapStateToProps = state => ({
  displayName: state.user.displayName,
  isContributor: state.user.isContributor
});

export default connect(mapStateToProps)(SignedIn);
