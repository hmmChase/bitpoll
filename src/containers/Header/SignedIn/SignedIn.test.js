import React from 'react';
import { shallow } from 'enzyme';
import { SignedIn, mapStateToProps } from './SignedIn';

describe('SignedIn', () => {
  let signedIn;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      displayName: 'Chase',
      isContributor: false,
      logOut: jest.fn()
    };
    signedIn = shallow(<SignedIn {...mockProps} />);
  });

  it('matches snapshot', () => {
    expect(signedIn).toMatchSnapshot();
  });

  it('matches snapshot with isContributor = true', () => {
    mockProps.isContributor = true;
    signedIn = shallow(<SignedIn {...mockProps} />);

    expect(signedIn).toMatchSnapshot();
  });

  it('calls props.logOut on click', () => {
    signedIn.find('.sign-out-btn').simulate('click');

    expect(mockProps.logOut).toHaveBeenCalledTimes(1);
  });

  describe('mapStateToProps', () => {
    it('maps state properties to props', () => {
      const mockState = {
        user: {
          displayName: 'Chase',
          isContributor: false
        }
      };
      const expected = {
        displayName: 'Chase',
        isContributor: false
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
});
