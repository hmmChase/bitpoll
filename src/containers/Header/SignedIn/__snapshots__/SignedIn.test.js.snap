// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignedIn matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    displayName="Chase"
    isContributor={false}
    logOut={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="log-in-out"
>
          <p
                    className="welcome-text"
          >
                    Welcome 
                    Chase
          </p>
</div>,
        <div
          className="sign-out-btn"
          onClick={[Function]}
>
          <span>
                    Log out
          </span>
</div>,
      ],
      "className": "signed-in-state",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="welcome-text"
>
              Welcome 
              Chase
</p>,
            false,
          ],
          "className": "log-in-out",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Welcome ",
                "Chase",
              ],
              "className": "welcome-text",
            },
            "ref": null,
            "rendered": Array [
              "Welcome ",
              "Chase",
            ],
            "type": "p",
          },
          false,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            Log out
</span>,
          "className": "sign-out-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Log out",
          },
          "ref": null,
          "rendered": "Log out",
          "type": "span",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="log-in-out"
>
            <p
                        className="welcome-text"
            >
                        Welcome 
                        Chase
            </p>
</div>,
          <div
            className="sign-out-btn"
            onClick={[Function]}
>
            <span>
                        Log out
            </span>
</div>,
        ],
        "className": "signed-in-state",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="welcome-text"
>
                Welcome 
                Chase
</p>,
              false,
            ],
            "className": "log-in-out",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Welcome ",
                  "Chase",
                ],
                "className": "welcome-text",
              },
              "ref": null,
              "rendered": Array [
                "Welcome ",
                "Chase",
              ],
              "type": "p",
            },
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              Log out
</span>,
            "className": "sign-out-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log out",
            },
            "ref": null,
            "rendered": "Log out",
            "type": "span",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SignedIn matches snapshot with isContributor = true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    displayName="Chase"
    isContributor={true}
    logOut={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="log-in-out"
>
          <p
                    className="welcome-text"
          >
                    Welcome 
                    Chase
          </p>
          <p
                    className="verifed-text"
          >
                    <FontAwesomeIcon
                              border={false}
                              className="verified-icon"
                              fixedWidth={false}
                              flip={null}
                              icon={
                                        Array [
                                                  "far",
                                                  "check-circle",
                                                ]
                              }
                              listItem={false}
                              mask={null}
                              name=""
                              pull={null}
                              pulse={false}
                              rotation={null}
                              size={null}
                              spin={false}
                              symbol={false}
                              transform={null}
                    />
                    verified contributor
          </p>
</div>,
        <div
          className="sign-out-btn"
          onClick={[Function]}
>
          <span>
                    Log out
          </span>
</div>,
      ],
      "className": "signed-in-state",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="welcome-text"
>
              Welcome 
              Chase
</p>,
            <p
              className="verifed-text"
>
              <FontAwesomeIcon
                            border={false}
                            className="verified-icon"
                            fixedWidth={false}
                            flip={null}
                            icon={
                                          Array [
                                                        "far",
                                                        "check-circle",
                                                      ]
                            }
                            listItem={false}
                            mask={null}
                            name=""
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size={null}
                            spin={false}
                            symbol={false}
                            transform={null}
              />
              verified contributor
</p>,
          ],
          "className": "log-in-out",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Welcome ",
                "Chase",
              ],
              "className": "welcome-text",
            },
            "ref": null,
            "rendered": Array [
              "Welcome ",
              "Chase",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FontAwesomeIcon
                  border={false}
                  className="verified-icon"
                  fixedWidth={false}
                  flip={null}
                  icon={
                                    Array [
                                                      "far",
                                                      "check-circle",
                                                    ]
                  }
                  listItem={false}
                  mask={null}
                  name=""
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  transform={null}
/>,
                "verified contributor",
              ],
              "className": "verifed-text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "verified-icon",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": Array [
                    "far",
                    "check-circle",
                  ],
                  "listItem": false,
                  "mask": null,
                  "name": "",
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "verified contributor",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            Log out
</span>,
          "className": "sign-out-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Log out",
          },
          "ref": null,
          "rendered": "Log out",
          "type": "span",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="log-in-out"
>
            <p
                        className="welcome-text"
            >
                        Welcome 
                        Chase
            </p>
            <p
                        className="verifed-text"
            >
                        <FontAwesomeIcon
                                    border={false}
                                    className="verified-icon"
                                    fixedWidth={false}
                                    flip={null}
                                    icon={
                                                Array [
                                                            "far",
                                                            "check-circle",
                                                          ]
                                    }
                                    listItem={false}
                                    mask={null}
                                    name=""
                                    pull={null}
                                    pulse={false}
                                    rotation={null}
                                    size={null}
                                    spin={false}
                                    symbol={false}
                                    transform={null}
                        />
                        verified contributor
            </p>
</div>,
          <div
            className="sign-out-btn"
            onClick={[Function]}
>
            <span>
                        Log out
            </span>
</div>,
        ],
        "className": "signed-in-state",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="welcome-text"
>
                Welcome 
                Chase
</p>,
              <p
                className="verifed-text"
>
                <FontAwesomeIcon
                                border={false}
                                className="verified-icon"
                                fixedWidth={false}
                                flip={null}
                                icon={
                                                Array [
                                                                "far",
                                                                "check-circle",
                                                              ]
                                }
                                listItem={false}
                                mask={null}
                                name=""
                                pull={null}
                                pulse={false}
                                rotation={null}
                                size={null}
                                spin={false}
                                symbol={false}
                                transform={null}
                />
                verified contributor
</p>,
            ],
            "className": "log-in-out",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Welcome ",
                  "Chase",
                ],
                "className": "welcome-text",
              },
              "ref": null,
              "rendered": Array [
                "Welcome ",
                "Chase",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FontAwesomeIcon
                    border={false}
                    className="verified-icon"
                    fixedWidth={false}
                    flip={null}
                    icon={
                                        Array [
                                                            "far",
                                                            "check-circle",
                                                          ]
                    }
                    listItem={false}
                    mask={null}
                    name=""
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    transform={null}
/>,
                  "verified contributor",
                ],
                "className": "verifed-text",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "verified-icon",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": Array [
                      "far",
                      "check-circle",
                    ],
                    "listItem": false,
                    "mask": null,
                    "name": "",
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": null,
                    "spin": false,
                    "symbol": false,
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "verified contributor",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              Log out
</span>,
            "className": "sign-out-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log out",
            },
            "ref": null,
            "rendered": "Log out",
            "type": "span",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
